target_chains/solana/programs/pyth-price-publisher/src/validate.rs

```diff
@@ -0,0 +1,115 @@
+use {
+    crate::{
+        accounts, ensure,
+        instruction::{CONFIG_SEED, PUBLISHER_CONFIG_SEED},
+    },
+    solana_program::{
+        account_info::AccountInfo, program_error::ProgramError, program_memory::sol_memcmp,
+        pubkey::Pubkey, system_program,
+    },
+};
+
+pub fn validate_publisher<'a>(
```

Comment:
probably not important but as i'm not so optimist on bpf optimizer I am wondering whether we could avoid clone and use reference.

Reply:
Removed clones.